pkg/bootstrap/node.go

```diff
@@ -188,35 +188,54 @@ func EnlargeStoragePool(baseImage string, nodes int) error {
 	// for the storage pool, every time when it pulls an image to store in
 	// the pool.
 	var extraSizeRatio float64 = 0.5
+	var err error
 
 	baseImageAbspath := path.Join(machinesDir, baseImage+".raw")
 
-	fipool, err := os.Stat(machinesImage)
-	if err != nil {
-		return err
+	if poolSize, err = getAllocatedFileSize(machinesImage); err != nil {
+		return 0, err
 	}
+	extraSize = int64(float64(poolSize) * extraSizeRatio)
 
-	poolSize = fipool.Size()
-
-	poolSize += int64(float64(fipool.Size()) * extraSizeRatio)
+	if biSize, err = getAllocatedFileSize(baseImageAbspath); err != nil {
+		return 0, err
+	}
+	extraSize += int64(float64(biSize)*extraSizeRatio) * int64(nodes)
 
-	fiBase, err := os.Stat(baseImageAbspath)
+	varDir, _ := path.Split(machinesImage)
+	freeVolSpace, err := getVolFreeSpace(varDir)
 	if err != nil {
-		return err
+		return 0, err
 	}
 
-	poolSize += int64(float64(fiBase.Size())*extraSizeRatio) * int64(nodes)
+	// extraSize, space to be allocated, shoud be 90% of freeVolSpace,
+	// actual free space on the target volume. Here 90% is simply a
+	// pre-defined estimation of how much space can be occupied.
+	// We should reserve some unallocated free space for the whole host,
+	// as 100% usage of rootfs could badly affect the system's reliability.
+	if extraSize >= int64((float64(freeVolSpace))*0.9) {
+		return 0, fmt.Errorf("no more free space in %s. Please get more space for creating %d nodes, each of which needs space of %d bytes.", varDir, nodes, biSize)
```

Comment:
This msg is not really accurate. There is free space because we haven't filled it with images because we detected that it would have taken all the free space. Let me suggest... "not enough space on disk for %d nodes. Each node needs about %d MB, so in total you'll need about %d MB available."